### PR TITLE
use Enum instead of validate_re for $sendfile

### PR DESCRIPTION
**Pull request by mmoll**

### CODE REVIEW

**Comment by ekohl** [comment]
I don't see why the tests failed.

**Comment by mmoll** [comment]
parallel_tests maybe? 😹

File: spec/classes/apache_spec.rb [hunk 1]
```diff
@@ -636,7 +636,7 @@
         it "should fail" do
           expect do
             catalogue
-          end.to raise_error(Puppet::Error, /"foo" does not match/)
+          end.to raise_error(Puppet::PreformattedError, /Evaluation Error: Error while evaluating a Resource Statement, Class\[Apache\]: parameter 'sendfile' expects a match for Enum\['Off', 'On', 'off', 'on'\], got 'foo'/)
```
Comment (bastelfreak) [inline spec/classes/apache_spec.rb]: is it still needed to test against those cases? In the Vox Pupuli namespace we dropped them

Comment (mmoll) [inline spec/classes/apache_spec.rb]: I can delete this for sure, if wanted...

File: manifests/init.pp [hunk 2]
```diff
@@ -38,7 +38,7 @@
   $purge_vhost_dir                                               = undef,
   $purge_vdir                                                    = false,
   $serveradmin                                                   = 'root@localhost',
-  $sendfile                                                      = 'On',
+  Enum['On', 'Off', 'on', 'off'] $sendfile                       = 'On',
```
Comment (TheMeier) [inline manifests/init.pp]: why not map On/Off, on/off to bool in puppet?

Comment (hunner) [inline manifests/init.pp]: Eh, overly complicated?

Comment (ekohl) [inline manifests/init.pp]: For simplification you could add a type ```Apache::OnOff``` since this validation is used in a lot of places.

**Comment by ekohl** [comment]
@mmoll if you rebase, do the tests pass then?

**Comment by mmoll** [comment]
Tests are green now.

File: manifests/init.pp (hunk 2 above)
Comment (mmoll) [inline manifests/init.pp]: I can do that, if wanted.

However, that would be a separate PR, which also changes all occurrences of it.

Comment (ekohl) [inline manifests/init.pp]: I think you're right they shouldn't be in this PR.

**Comment by mmoll** [comment]
anything needed to get this in?

**Pull request merged**